file_transfer_agent_test.go

```diff
@@ -661,3 +661,96 @@ func TestReadonlyTmpDirPathShouldFail(t *testing.T) {
 		t.Fatalf("should not upload file as temporary directory is not readable")
 	}
 }
+
+func TestUploadDownloadOneFileRequireCompress(t *testing.T) {
+	testUploadDownloadOneFile(t, false)
+}
+
+func TestUploadDownloadOneFileRequireCompressStream(t *testing.T) {
+	testUploadDownloadOneFile(t, true)
+}
+
+func testUploadDownloadOneFile(t *testing.T, isStream bool) {
+	tmpDir, err := os.MkdirTemp("", "data")
```

Comment:
Can we use `t.tempDir`? In that case we don't have to worry about cleaning up the data.

Reply:
`t.tempDir` fails on windows so I'll revert it back to use `os.MkdirTemp`
> testing.go:1206: TempDir RemoveAll cleanup: remove C:\Users\RUNNER~1\AppData\Local\Temp\TestUploadDownloadOneFileRequireCompressStream1571942357\001\data.txt: The process cannot access the file because it is being used by another process.